vnext/Microsoft.ReactNative/RedBox.cpp

```diff
@@ -140,9 +144,19 @@ struct RedBox : public std::enable_shared_from_this<RedBox> {
 
  private:
   void UpdateErorrMessageUI() noexcept {
-    std::regex colorsreg("\\x1b\\[[0-9;]*m"); // strip out console colors which is often added to JS error messages
-    m_errorMessageText.Text(
-        Microsoft::Common::Unicode::Utf8ToUtf16(std::regex_replace(m_errorInfo.Message, colorsreg, "")));
+    const std::regex colorsRegex("\\x1b\\[[0-9;]*m"); // strip out console colors which is often added to JS error messages
+    std::string formated = std::regex_replace(m_errorInfo.Message, colorsRegex, "");
+
+    auto json = folly::parseJson(m_errorInfo.Message);
+    if (json.count("name") && json["name"] == "Error") {
+      auto message = json["message"].asString();
+      auto stack = json["stack"].asString().substr(ARRAYSIZE("Error: ") + message.length());
+      m_errorMessageText.Text(Microsoft::Common::Unicode::Utf8ToUtf16(message));
+      m_errorStackText.Text(Microsoft::Common::Unicode::Utf8ToUtf16(stack));
+    } else {
+      m_errorMessageText.Text(
+          Microsoft::Common::Unicode::Utf8ToUtf16(formated));
+    }
```

Comment:
Do you have to clear m_errorStackText or is this guaranteed to be fresh each time?